build-system/compile/check-for-unknown-deps.js

```diff
@@ -15,38 +15,39 @@
  */
 'use strict';
 
-const through = require('through2');
+const argv = require('minimist')(process.argv.slice(2));
+const fs = require('fs-extra');
 const {log} = require('../common/logging');
 const {red, cyan, yellow} = require('kleur/colors');
 
 /**
  * Searches for the identifier "module$", which Closure uses to uniquely
  * reference module imports. If any are found, that means Closure couldn't
  * import the module correctly.
- *
- * @return {!NodeJS.ReadStream}
+ * @param {string} file
+ * @return {Promise<void>}
  */
-exports.checkForUnknownDeps = function () {
+async function checkForUnknownDeps(file) {
   const regex = /[\w$]*module\$[\w$]+/;
-
-  return through.obj(function (file, _encoding, cb) {
-    const contents = file.contents.toString();
-    if (!contents.includes('module$')) {
-      // Fast check, since regexes can backtrack like crazy.
-      return cb(null, file);
-    }
-
-    const match = regex.exec(contents) || [
-      `couldn't parse the dep. Look for "module$" in the file`,
-    ];
-
-    log(
-      red('Error:'),
-      `Unknown dependency ${cyan(match[0])} found in ${cyan(file.relative)}`
-    );
+  const contents = await fs.readFile(file, 'utf-8');
+  if (!contents.includes('module$')) {
+    // Fast check, since regexes can backtrack like crazy.
+    return;
+  }
+  const match = regex.exec(contents) || [
+    `couldn't parse the dep. Look for "module$" in the file`,
+  ];
+  log(
+    red('Error:'),
+    `Unknown dependency ${cyan(match[0])} found in ${cyan(file)}`
+  );
+  if (argv.debug) {
+    log(red('Output file contents:'));
     log(yellow(contents));
-    const err = new Error('Compilation failed due to unknown dependency');
-    err.showStack = false;
```

Comment:
I really disliked the `showStack` omitting the stack frame of the build-system call that threw the error, and I'm happy it's dead.

Reply:
Yes, `showStack` was needed once upon a time because `gulp` would spam the console with a large mostly meaningless call stack. It's going away soon. (Assuming I can get this PR to land!)